### PR TITLE
Add: GitHub Actions for CI/CD

### PR DESCRIPTION
**Pull request by Kevin-Lee**
Add: GitHub Actions for CI/CD

https://github.com/Kevin-Lee/scala-hedgehog/actions

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -41,7 +42,7 @@ lazy val core = Project(
 lazy val example = Project(
     id = "example"
   , base = file("example")
-  ).settings(standardSettings ++ Seq(
+  ).settings(standardSettings ++ noPublish ++ Seq(
```
Comment (Kevin-Lee) [inline build.sbt]: I think this one should be a `noPublish` one.

File: .github/workflows/sbt-build.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  then
+    echo "Scala version is missing. Please enter the Scala version."
+    echo "sbt-build-simple.sh 2.11.12"
+    exit 1
+else
+  SCALA_VERSION=$1
+  CURRENT_BRANCH_NAME="${GITHUB_REF#refs/heads/:-}"
+  if [[ "$CURRENT_BRANCH_NAME" == "master" ]]
+  then
+    sbt -J-Xmx2048m "; ++ ${SCALA_VERSION}!; clean; test; packagedArtifacts"
```
Comment (Kevin-Lee) [inline .github/workflows/sbt-build.sh]: For `master` branch, it does `packagedArtifacts` to check it does not fail in packaging doc and source.
I don't think you want to see it broken after you create a tag.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+name: Build All
+
+on: [push]
+
+jobs:
+
+  build_scala2_10:
```
Comment (nightscape) [inline .github/workflows/build.yml]: Why not use a build matrix strategy instead of 4 copies of the same job?

Comment (Kevin-Lee) [reply .github/workflows/build.yml]: That would be great if I can use it. I need `2.10` for caching and `2.10.7` for sbt build. Do you know if I can have pairs in matrix?

Comment (Kevin-Lee) [reply .github/workflows/build.yml]: Actually I have an idea. I'll test it with my personal project first. Thanks for your suggestion to use matrix.

Comment (Kevin-Lee) [reply .github/workflows/build.yml]: It looks good.
https://github.com/Kevin-Lee/scala-hedgehog/commit/553054fdc20cdaaedc79060ae5a294b9588bf84e/checks

**Comment by Kevin-Lee** [comment]
Weird. It looks like Travis build is randomly broken. I guess it's good time to switch to GitHub Actions. 😄

**Comment by Kevin-Lee** [comment]
GitHub Actions is fine.
https://github.com/Kevin-Lee/scala-hedgehog/commit/11cad32026b9d5c3f5b71346b81521365113c5b7/checks

**Comment by Kevin-Lee** [comment]
Now it's fine.

**Comment by Kevin-Lee** [comment]
I've talked about this PR with @charleso and am merging this PR on behalf of Charles.

**Comment by Kevin-Lee** [comment]
I'm in charge of fixing the broken sync with Maven Central and will fix it with GitHub Actions.

**Pull request merged**